Src/AutoFixture/Kernel/TypeEnvy.cs

```diff
@@ -11,17 +11,12 @@ internal static class TypeEnvy
         public static TAttribute GetAttribute<TAttribute>(object candidate)
             where TAttribute : Attribute
         {
-            var member = candidate as MemberInfo;
-            if(member != null)
-            {
-                return member.GetCustomAttribute<TAttribute>();
-            }
-            var parameter = candidate as ParameterInfo;
-            if(parameter != null)
-            {
-                return parameter.GetCustomAttribute<TAttribute>();
-            }
-            return null;
+            var attributeProvider = candidate as ICustomAttributeProvider;
+
+            return attributeProvider?
+                .GetCustomAttributes(typeof(TAttribute), true)
+                .Cast<TAttribute>()
```

Comment:
I'm not exactly familiar with how this method is used now, but it's worth briefly considering the performance impact of LINQ statements.